src/components/view/pages/main/mainPageView.ts

```diff
@@ -0,0 +1,120 @@
+            <div class="store-page">
+              <section class="main-catalog">
+                <div class="main-catalog__filters">
+                  ${this.views.filters.render(data.filters)}
+                </div>
+                <div class="main-catalog__center-section main-center-section">
+                  ${this.views.sorting.render(data.view)}
+                  </div>
+                  <div class="main-catalog__products ${data.switchType === 'row' ? 'row-view' : ''}">
+                    ${this.views.productList.render({
+                      cart: data.cart,
+                      products: data.products
+                    })}
+                  </div>
+                  ${this.views.paginationPages.render({
+                    selectedPage: data.view.pagination.page,
+                    pageCount: data.view.pagination.pageCount,
+                  })}
+                </div>
+              </section>
+            </div>
+          </main>
+          ${this.views.footer.render()}
+        `
+    }
+
+    public afterRender(controller: Controller): void {
+        super.afterRender(controller);
+
+        const switchingRow = document.querySelector('.switch-view__line') as HTMLElement;
+        const switchingBlock = document.querySelector('.switch-view__block') as HTMLElement;
+        const mainCatalogProducts = document.querySelector('.main-catalog__products') as HTMLElement;
+        const inputSearch = document.querySelector('.find-input') as HTMLInputElement;
+        const clearButton = document.querySelector('.find-input-img_clear') as HTMLElement;
+
+        switchingRow.addEventListener('click', () => {
+            Router.setUrlParam('switch-view', 'row')
+        })
+
+        switchingBlock.addEventListener('click', () => {
+            Router.setUrlParam('switch-view', 'block')
+        })
+
+        if (mainCatalogProducts.classList.contains('row-view')) {
+            switchingRow.classList.add('switch-active');
+            switchingBlock.classList.remove('switch-active');
+        }
+
+        clearButton.addEventListener('click', () => {
+            Router.removeUrlParamKey('q')
+        })
+
+        let timeouts: TimeoutId[] = []
+        inputSearch.addEventListener('input', () => {
+            timeouts.forEach((timeout) => clearTimeout(timeout))
+            timeouts.push(
+                setTimeout(() => {
+                    timeouts.push(
+                        setTimeout(() => {
+                            const newInputSearch = document.querySelector('.find-input') as HTMLInputElement;
+                            newInputSearch.focus();
+                            newInputSearch.setSelectionRange(inputSearch.selectionStart, inputSearch.selectionEnd)
+                        }, 1)
+                    );
+
+                    Router.removeUrlParamKey('page');
+                    inputSearch.value ? Router.setUrlParam('q', inputSearch.value) : Router.removeUrlParamKey('q');
```

Comment:
Please do not use `? : ` in such kind of cases
use `if`
`if (inputSearch.value) {
Router.setUrlParam('q', inputSearch.value)
} else {
Router.removeUrlParamKey('q')
}`